hudi-common/src/main/java/org/apache/hudi/metadata/MetadataPartitionType.java

```diff
@@ -319,6 +331,16 @@ public boolean isMetadataPartitionAvailable(HoodieTableMetaClient metaClient) {
     this.recordType = recordType;
   }
 
+  /**
+   * Get the partition name from the metadata partition type.
+   * NOTE: For certain types of metadata partition, such as functional index and secondary index,
+   * partition path defined enum is just the prefix to denote the type of metadata partition.
+   * The actual partition name is contained in the index definition.
+   */
+  public String getPartitionPath(HoodieTableMetaClient metaClient, String indexName) {
+    return getPartitionPath();
+  }
+
   public String getPartitionPath() {
     return partitionPath;
   }
```

Comment:
We should remove this API now.

Reply:
It's still being used at some places where we don't have metaclient or indexName. Technically, this can be removed by making the other method accept `Option<HoodieTableMetaClient> metaClient, Option<String> indexName` as arguments, but it will change a lot of places. If you're ok, I can do it in a separate PR after landing this one (just to keep this PR a bit clean).

Reply:
Makes sense.

Reply:
Let's have a JIRA to track it.